### PR TITLE
DEP Upgrade frontend build stack

### PR DESCRIPTION
**Pull request by GuySartorelli**
**DO NOT SQUASH** - the commits in this PR are distinct.

## Parent issue
- https://github.com/silverstripeltd/product-issues/issues/644

### CODE REVIEW

**Comment by GuySartorelli** [comment]
unit test failure is unrelated

File: .eslintignore [hunk 1]
```diff
@@ -1,8 +1,5 @@
 # Ignore dist files
 client/dist/
 
-# Ignore legacy files
-client/src/legacy/
```
Comment (ChloeSartorelli) [inline .eslintignore:5]: It's not _actually_ legacy - we're not getting rid of it any time soon so it should meet our linting standards.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Got this javascript error when manually testing ticking a submitted comment in the comments tab and then select 'Approve' from the dropdown

image

I do have `colymba/gridfield-bulk-editing-tools` `dev-pulls/4/frontend-build-stack` if that makes a difference

**Comment by GuySartorelli** [comment]
The comments form currently isn't rendering - https://github.com/silverstripe/silverstripe-framework/pull/10676 needs to be merged to fix that.

**Comment by GuySartorelli** [comment]
Email address validation also broken - https://github.com/silverstripe/comment-notifications/pull/50 fixes that

**Comment by GuySartorelli** [comment]
The error you mentioned is now fixed in the bulk editing tools PR

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**